app/controllers/pieces_controller.rb

```diff
@@ -4,22 +4,24 @@ def update
     @piece = Piece.find(params[:id])
     @game = Game.find(@piece.game_id)
 
-    if @piece.valid_move?(params[:y_coord].to_i, params[:x_coord].to_i)
+    if @piece.valid_move?(params[:y_coord].to_i, params[:x_coord].to_i) && !@piece.moving_into_check?(params[:y_coord].to_i, params[:x_coord].to_i) 
```

Comment:
@xelzi, I agree, I think we should call check? beforehand to prevent someone from moving a piece that doesn't remove them from check